src/core/event-handle.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Event Handle that is created by {@link EventHandler} and can be used for easier removing and events management.
+ * @example
+ * const evt = obj.on('test', (a, b) => {
+ *     console.log(a + b);
+ * });
+ * obj.fire('test');
+ *
+ * evt.off(); // easy way to remove this event
+ * obj.fire('test'); // this will not trigger an event
+ * @example
+ * // store an array of event handles
+ * let events = [ ];
+ *
+ * events.push(objA.on('testA', () => { }));
+ * events.push(objB.on('testB', () => { }));
+ *
+ * // when needed, remove all events
+ * events.forEach((evt) => {
+ *     evt.off();
+ * });
+ * events = [ ];
+ */
+class EventHandle {
+    /**
+     * @type {import('./event-handler.js').EventHandler}
+     * @private
+     */
+    handler;
+
+    /**
+     * @type {string}
+     * @private
+     */
+    name;
+
+    /**
+     * @type {import('./event-handler.js').HandleEventCallback}
+     * @ignore
+     */
+    callback;
+
+    /**
+     * @type {object}
+     * @ignore
+     */
+    scope;
+
+    /**
+     * @type {boolean}
+     * @ignore
+     */
+    once;
+
+    /**
+     * True if event has been removed.
+     * @type {boolean}
+     * @private
+     */
+    _removed = false;
+
+    /**
+     * @param {import('./event-handler.js').EventHandler} handler - source object of the event.
+     * @param {string} name - Name of the event.
+     * @param {import('./event-handler.js').HandleEventCallback} callback - Function that is called when event is fired.
+     * @param {object} scope - Object that is used as `this` when event is fired.
+     * @param {boolean} [once] - If this is a single event and will be removed after event is fired.
+     */
+    constructor(handler, name, callback, scope, once = false) {
+        this.handler = handler;
+        this.name = name;
+        this.callback = callback;
+        this.scope = scope;
+        this.once = once;
+    }
+
+    /**
+     * Remove references.
+     * @ignore
+     */
+    destroy() {
+        if (this._removed) return;
+        this._removed = true;
+    }
+
+    /**
+     * Remove this event from its handler.
+     */
+    off() {
+        if (this._removed) return;
+        this.handler.off(this.name, this.callback, this.scope);
+    }
+
+    /**
+     * True if event has been removed.
+     * @type {boolean}
+     */
+    get removed() {
+        return this._removed;
+    }
```

Comment:
Do we really need all this extra getter code which kinda does nothing? Why not simple and short:
```js
    /**
     * True if event has been removed.
     * @type {boolean}
     * @readonly
     */
    removed = false;
```

If I get your intention right, you just want to have it readable and not writable?

Reply:
Sorry, I just realized we still lack proper types to express this idiom: https://github.com/microsoft/TypeScript/issues/37487

My favorite would be this, which is a pattern all over the engine:

```js
    destroy() {
        this.handler?.off(this.name, this.callback, this.scope);
        this.handler = null;
    }
```

So we don't introduce `removed`/`_removed` at all.